benchmarks/src/bin/tpch.rs

```diff
@@ -1611,65 +1611,137 @@ mod tests {
                     "optimized logical plan round trip failed"
                 );
+            }
+
+            Ok(())
+        }
+
+        async fn round_trip_physical_plan(n: usize) -> Result<()> {
+            let config = SessionConfig::new()
+                .with_target_partitions(1)
+                .with_batch_size(10);
+            let ctx = SessionContext::with_config(config);
+            let session_state = ctx.state();
+            let codec: BallistaCodec<
+                datafusion_proto::protobuf::LogicalPlanNode,
+                datafusion_proto::protobuf::PhysicalPlanNode,
+            > = BallistaCodec::default();
+
+            // set tpch_data_path to dummy value and skip physical plan serde test when TPCH_DATA
+            // is not set.
+            let tpch_data_path =
+                env::var("TPCH_DATA").unwrap_or_else(|_| "./".to_string());
+            let path = ListingTableUrl::parse(tpch_data_path)?;
+
+            for &table in TABLES {
+                let schema = get_schema(table);
+                let options = CsvReadOptions::new()
+                    .schema(&schema)
+                    .delimiter(b'|')
+                    .has_header(false)
+                    .file_extension(".tbl");
+                let cfg = SessionConfig::new();
+                let listing_options = options.to_listing_options(&cfg);
+                let config = ListingTableConfig::new(path.clone())
+                    .with_listing_options(listing_options)
+                    .with_schema(Arc::new(schema));
+                let provider = ListingTable::try_new(config)?;
+                ctx.register_table(table, Arc::new(provider))?;
+            }
+
+            // test logical plan round trip
+            let plans = create_logical_plans(&ctx, n).await?;
+            for plan in plans {
+                let plan = session_state.optimize(&plan)?;
 
                 // test physical plan roundtrip
-                if env::var("TPCH_DATA").is_ok() {
-                    let physical_plan = session_state.create_physical_plan(&plan).await?;
-                    let proto: datafusion_proto::protobuf::PhysicalPlanNode =
-                        datafusion_proto::protobuf::PhysicalPlanNode::try_from_physical_plan(
-                            physical_plan.clone(),
-                            codec.physical_extension_codec(),
-                        )
-                        .unwrap();
-                    let runtime = ctx.runtime_env();
-                    let round_trip: Arc<dyn ExecutionPlan> = proto
-                        .try_into_physical_plan(
-                            &ctx,
-                            runtime.deref(),
-                            codec.physical_extension_codec(),
-                        )
-                        .unwrap();
-                    assert_eq!(
-                        format!("{physical_plan:?}"),
-                        format!("{round_trip:?}"),
```

Comment:
debug output is non-deterministic (alias maps print in different order) so changed to use `displayable` instead

Reply:
Makes sense...